homeassistant/components/smartthings/config_flow.py

```diff
@@ -87,13 +86,17 @@ def __init__(self):
                 app = await create_app(self.hass, self.api)
             setup_smartapp(self.hass, app)
             self.app_id = app.app_id
-        except ClientResponseError as ex:
+        except APIResponseError as ex:
             if ex.status == 401:
                 errors[CONF_ACCESS_TOKEN] = "token_unauthorized"
             elif ex.status == 403:
                 errors[CONF_ACCESS_TOKEN] = "token_forbidden"
+            elif ex.is_target_error():
+                errors['base'] = 'webhook_error'
             else:
                 errors['base'] = "app_setup_error"
+            _LOGGER.exception("Unexpected error setting up the SmartApp: %s",
```

Comment:
Shouldn't this be indented under the `else` case?

Reply:
Yes and actually you pointing this out made me realize a different mistake -- `APIResponseError` only encapsulates 400, 422, 429, and 500 -- so the original handling for 401 and 403 is still needed.  I added new `except` clause for `APIResponseError` and we do always want to log the response, even if it is a `webhook` error, as this will contain the details of what SmartThings is wrong.